src/modules/fancyzones/editor/FancyZonesEditor/App.xaml.cs

```diff
@@ -42,59 +42,82 @@ public partial class App : Application
         private const string CrashReportLocationNullTag = "location is null or empty";
 
+        // Localizable strings
+        private const string ErrorMessageBoxTitle = "FancyZones Editor Exception Handler";
+        private const string ErrorMessageBoxMessage = "Please report the bug to ";
+
         private readonly IFileSystem _fileSystem = new FileSystem();
 
-        public Settings ZoneSettings { get; }
+        public MainWindowSettingsModel MainWindowSettings { get; }
+
+        public static FancyZonesEditorIO FancyZonesEditorIO { get; private set; }
+
+        public static Overlay Overlay { get; private set; }
+
+        public static int PowerToysPID { get; set; }
+
+        public static bool DebugMode
+        {
+            get
+            {
+                return _debugMode;
+            }
+        }
+
+        private static bool _debugMode = false;
+
+        [Conditional("DEBUG")]
+        private void DebugModeCheck()
+        {
+            _debugMode = true;
+        }
 
         public App()
         {
-            ZoneSettings = new Settings();
+            DebugModeCheck();
+            FancyZonesEditorIO = new FancyZonesEditorIO();
+            Overlay = new Overlay();
+            MainWindowSettings = new MainWindowSettingsModel();
         }
 
         private void OnStartup(object sender, StartupEventArgs e)
         {
             AppDomain.CurrentDomain.UnhandledException += OnUnhandledException;
 
-            RunnerHelper.WaitForPowerToysRunner(Settings.PowerToysPID, () =>
+            RunnerHelper.WaitForPowerToysRunner(PowerToysPID, () =>
             {
                 Environment.Exit(0);
             });
 
-            LayoutModel foundModel = null;
+            FancyZonesEditorIO.ParseCommandLineArguments();
+            FancyZonesEditorIO.ParseDeviceInfoData();
 
-            foreach (LayoutModel model in ZoneSettings.DefaultModels)
-            {
-                if (model.Type == Settings.ActiveZoneSetLayoutType)
-                {
-                    // found match
-                    foundModel = model;
-                    break;
-                }
-            }
+            MainWindowSettingsModel settings = ((App)Current).MainWindowSettings;
+            settings.UpdateSelectedLayoutModel();
 
-            if (foundModel == null)
-            {
-                foreach (LayoutModel model in Settings.CustomModels)
-                {
-                    if ("{" + model.Guid.ToString().ToUpper() + "}" == Settings.ActiveZoneSetUUid.ToUpper())
-                    {
-                        // found match
-                        foundModel = model;
-                        break;
-                    }
-                }
-            }
+            Overlay.Show();
+        }
 
-            if (foundModel == null)
+        public static void ShowExceptionMessageBox(string message, Exception exception = null)
```

Comment:
these should probably get their own file/class

Reply:
We will have to refactor most of the code after porting to .NET Core and introducing the new UI, so I agree but we can postpone it.